postgresql/resource_postgresql_grant.go

```diff
@@ -442,7 +442,9 @@ func checkRoleDBSchemaExists(client *Client, d *schema.ResourceData) (bool, erro
 		return false, nil
 	}
 
-	if d.Get("object_type").(string) != "database" {
+	pgSchema := d.Get("schema").(string)
+
+	if d.Get("object_type").(string) != "database" && pgSchema != "" {
```

Comment:
Is it done on purpose? For `postgresql_grant`, `schema` is still required when `object_type` is not database.

Reply:
The `resourcePostgreSQLDefaultPrivilegesRead` function calls the function `checkRoleDBSchemaExists` to validate that the role, the database and the schema exists. Obviously if the schema is blank then it doesnt need to be validated.

This validation function is shared between 2 resource types. TBH I wasnt sure if this could break something for the `postgresql_grant` resource type but the acceptance tests all seemed to pass.

One option is to make a similar function without the schema validation called checkRoleDBExists. The `resourcePostgreSQLDefaultPrivilegesRead` function can conditionally call this one depending whether or not the default privileges are being granted on a schema. 
This way we can be sure that there are not edge cases lurking out there that run into issues with the change that you highlighted.
What do you think?

Reply:
> The resourcePostgreSQLDefaultPrivilegesRead function calls the function checkRoleDBSchemaExists to validate that the role

Ho yes I forgot that :+1: 

>  What do you think?

I think we can keep this function like that but to avoid the edge case we can put back the validation [you've just removed](https://github.com/cyrilgdn/terraform-provider-postgresql/pull/59/files#diff-5b6507de6d12f48a7e2b0ab86628e2ff023bf030d2e3727692e8b1caf3de501fL257-L261) directly in the `resourcePostgreSQLGrantCreate`.

Reply:
Ah, OK. I think I follow you. I have pushed another change to implement this. Let me know if this covers it.